odoo/tools/translate.py

```diff
@@ -99,17 +100,25 @@ def is_translatable_attrib_text(node):
 
 # regexpr for string formatting and extract ( ruby-style )|( jinja-style  ) used in `_compile_format`
 FORMAT_REGEX = re.compile(r'(?:#\{(.+?)\})|(?:\{\{(.+?)\}\})')
+VARNAME_REGEXP = re.compile(r'^[A-Za-z_][A-Za-z0-9_]*$')
+
 
 def translate_format_string_expression(term, callback):
     expressions = {}
-    def add(exp_py):
-        index = len(expressions)
-        expressions[str(index)] = exp_py
-        return '{{%s}}' % index
-    term_without_py = FORMAT_REGEX.sub(lambda g: add(g.group(0)), term)
+    index = itertools.count()
+    def add(g):
+        key = next(index)
+        expressions[str(key)] = g.group(0)
+        expression = g.group(1) or g.group(2)
+        if VARNAME_REGEXP.match(expression):
+            expressions[expression] = g.group(0)
+            return '{{%s}}' % expression
+        else:
+            return '{{%s}}' % key
+    term_without_py = FORMAT_REGEX.sub(lambda g: add(g), term)
     translated_value = callback(term_without_py)
     if translated_value:
-        return FORMAT_REGEX.sub(lambda g: expressions.get(g.group(0)[2:-2], 'None'), translated_value)
+        return FORMAT_REGEX.sub(lambda g: expressions.get(g.group(1) or g.group(2), "{{'WRONG TRANSLATION'}}"), translated_value)
```

Comment:
It doesn't sound like a good idea to put `{{'WRONG TRANSLATION'}}` if the translation isn't correct. 
Putting a empty string seems less aggressive, right ?

IMO, we should add a warning with the lang + term that are not correct. What do you think :thinking: ?